deployment/execution.js

```diff
@@ -7,8 +7,10 @@ const fs = require("fs");
  * @param {string} template - Template location that will be used to create subgraph.yaml
  * @param {string} location - Location in the subgraph will be deployed to {e.g. messari/uniswap-v2-ethereum}
  */
-function scripts(protocol, network, template, location, constants) {
+function scripts(protocol, network, template, location, constants, type) {
```

Comment:
I do a check for a null value already, which it then assumes deploy. This would still require a null check though wouldn't it? Because I would still be passing the argument.

Reply:
Oh, higher up the stack? This function is never called with a null `type` then, right? Even if the argument is left empty in the package.json command

Reply:
Well - its a bit confusing, but it checks if it is a valid entry at the beginning - either null, build, or deploy. Then if the value is null or deploy it just deploys, build if build.